dev_scripts_helpers/github/sync_gh_issue_labels.py

```diff
@@ -0,0 +1,326 @@
+        :param label: label object to create
+        """
+        url = f"https://api.github.com/repos/{owner}/{repo}/labels"
+        data = {
+            "name": label.name,
+            "description": label.description,
+            "color": label.color,
+        }
+        response = requests.post(url, headers=self.headers, json=data)
+        response.raise_for_status()
+        _LOG.info("Label created: %s", label)
+
+    def update_label(self, owner: str, repo: str, label: Label) -> None:
+        """
+        Update an existing label in the repository.
+
+        :param owner: GitHub repository owner/organization
+        :param repo: GitHub repository name
+        :param label: label object to update
+        """
+        url = f"https://api.github.com/repos/{owner}/{repo}/labels/{label.name}"
+        data = {
+            "name": label.name,
+            "description": label.description,
+            "color": label.color,
+        }
+        response = requests.patch(url, headers=self.headers, json=data)
+        response.raise_for_status()
+        _LOG.info("Label updated: %s", label)
+
+    def delete_label(self, owner: str, repo: str, name: str) -> None:
+        """
+        Delete a label from the repository.
+
+        :param owner: GitHub repository owner/organization
+        :param repo: GitHub repository name
+        :param name: label name to delete
+        """
+        url = f"https://api.github.com/repos/{owner}/{repo}/labels/{name}"
+        response = requests.delete(url, headers=self.headers)
+        response.raise_for_status()
+        _LOG.info("Label deleted: %s", name)
+
+
+def load_labels(path: str) -> List[Label]:
+    """
+    Load labels from label inventory manifest file.
+
+    :param path: path to label inventory manifest file
+    :return: label objects
+    """
+    try:
+        with open(path, "r") as file:
+            yaml_data = yaml.safe_load(file)
+            labels = [
+                Label(
+                    name=item["name"],
+                    description=item["description"],
+                    color=item["color"],
+                )
+                for item in yaml_data
+            ]
+            return labels
+    except Exception as e:
+        _LOG.error("Error loading label inventory manifest file: %s", str(e))
+        sys.exit(1)
+
+
+def save_labels(labels: List[Label], path: str) -> None:
+    """
+    Save labels to the label inventory manifest file.
+
+    :param labels: label objects
+    :param path: path to save the label inventory manifest file to
+    """
+    try:
+        with open(path, "w") as file:
+            yaml_data = [
+                {
+                    "name": label.name,
+                    "description": label.description,
+                    "color": f"#{label.color}",
+                }
+                for label in labels
+            ]
+            yaml.dump(yaml_data, file, default_flow_style=False, sort_keys=False)
```

Comment:
We should comment the magic values, e.g., `default_flow_style`. Why are we setting to false?